### ISSUE TITLE
Single- & MultiSelectField: Loading icon not consistent with InputField

### ISSUE THREAD

**Issue by Mohammer5** [dhis2/ui#248]
When passing the `loading` prop to the Single- & MultiSelectField, the loading indicator that's shown on the InputField is missing here. I think we should add it for consistency reasons

CC: @cooper-joe

**Comment by cooper-joe** [comment]
I remember discussing the loading state of `select` – the main point was whether to show visible loading state when items were loading. This is slightly different than the loading state of an `input`, which only has it's own value to load.

The following system makes sense to me:
- Loading/validating/doing something with the selected value: show loading spinner next to the field (like `input`)
- Loading the available values: show loading spinner inside the menu when it is opened.

So, I agree. The `select`s should get this loading state, but it should not be used when loading available items in a menu as we have a separate loading state for that.

**Comment by Mohammer5** [comment]
> but it should not be used when loading available items in a menu as we have a separate loading state for that.

This would be confusing behavior to me, I'll explain:

#### The select has no items and items are being loaded

On the one hand, if nothing is shown, I'd expect the select to work. I'll only know that the items are being loaded after I click on the select.

On the other hand, while no options are available, the select is not functional at all, so I'd rather disable it (cursor: not-allowed, not the gray background) while it's loading to indicate that this field can't be used yet.

I assume the same behavior would make sense for loading/validating/doing something with the selected value.

#### The select has items but more are being loaded

I would expect the loading indicator to be inside the dropdown, below the list of options.

<hr />

So basically the loading indicator on the outside means: You can't use this field right now
The loading indicator inside means: We're loading more options, but you can already pick one of the listed options

**Comment by cooper-joe** [comment]
I see the point here, but one of my concerns is displaying a loading spinner outside of a select (next to the field), when the user has not interacted with it. A loading spinner for the user communicates "options for this select are loading" - but I would argue that this is not necessarily relevant information for a user. 

If a user loads a page with 8 different selects in different cards/panels/areas, seeing 8 loading spinners does not really give the user valuable information because the user has not interacted with those selects and therefore has no reason to expect feedback from them. 

Therefore we need to choose between two scenario that have the potential to be confusing:
1. **Loading spinner inside the menu when loading initial options**: as you mentioned above, "if nothing is shown, I'd expect the select to work. I'll only know that the items are being loaded after I click on the select."
2. **Loading spinner outside of the menu when loading initial options**:  as I wrote above, the user would see loading spinners without interacting with items which could lead to the "what did I do? Why are these fields spinning when I haven't touched them?".

I argue that the problem raised by 1. is an inconvenience that would in the worst case lead to a user thinking: "I wish I knew this select was loading before I clicked on it". The worst case in scenario 2. is that a user is confused by an interface that is showing feedback without their input, thus weakening the idea that "my actions have a reaction" that supports good UX.

---

A potential middle ground would be to use a placeholder text inside the select field that reads "Loading..." on initial load. This reduces the visual impact that the spinner would add, while still providing feedback without needing to open the select first. The select could also get `cursor: not-allowed` during this time.

![image](https://user-images.githubusercontent.com/33054985/91567168-fecb3680-e944-11ea-8d8d-e4ee3c8d75c6.png)

**Comment by Mohammer5** [comment]
> Therefore we need to choose between two scenario that have the potential to be confusing:

I see, fair enough!

> A potential middle ground would be to use a placeholder text inside the select field that reads "Loading..." on initial load. This reduces the visual impact that the spinner would add, while still providing feedback without needing to open the select first. The select could also get `cursor: not-allowed` during this time.

This sounds good to me! :ok_hand:

**Comment by ismay** [comment]
I think I'd prefer the original behaviour in this scenario. If you have a page with a lot of selects that are loading options, most likely you'll see a ton of `Loading options...` flash across the screen without being able to read all of them. I don't think that as a user I'd be very interested in whether they're loading items, unless I'm opening the select.

Keeping the loading state in the menu for when it's loading options seems like it'd make for a calmer, less confusing UI to me.

**Comment by Mohammer5** [comment]
Closing. I guess we could either use @ismay's proposal to use the current state or re-open this issue again
